modules/redis/redis.c

```diff
@@ -221,6 +207,26 @@ redis_dd_disconnect(LogThreadedDestDriver *s)
   self->c = NULL;
 }
 
+static void
+_fill_argv(gpointer data, gpointer user_data)
+{
+  RedisDriver *self = ((gpointer *)user_data)[0];
+  LogMessage *msg = ((gpointer *)user_data)[1];
+  const char **argv = ((gpointer *)user_data)[2];
+  size_t *argvlen = ((gpointer *)user_data)[3];
+  gint *index = ((gpointer *)user_data)[4];
```

Comment:
This is a lot of user data. Since you need the index as well (and `g_list_foreach` won't give you the index) , I'd transform this into an ordinary loop.

Reply:
Done